app/views/products/new.html.erb

```diff
@@ -0,0 +1,9 @@
+<section>
+    <h2><%= form_for [@vendor, @product] do |f| %>
+      <%= f.label :name %>
+      <span class="field"><%= f.text_field :name %></span>
+      <%= f.hidden_field(:vendor_id, :value => @vendor.id)  %>
+
+      <%= f.submit 'Create Product' %>
+    <% end %></h2>
```

Comment:
This would be easier to read if the h2 opening and closing tags were on their own lines and the form lines were indented: 
`    <h2>
          <%= form_for [@vendor, @product] do |f| %>
              <%= f.label :name %>
              <span class="field"><%= f.text_field :name %></span>
              <%= f.hidden_field(:vendor_id, :value => @vendor.id)  %>
              <%= f.submit 'Create Product' %>
          <% end %>
     </h2>`